integrations/oci/metrics/metrics/src/test/java/io/helidon/integrations/oci/metrics/OciMetricsSupportTest.java

```diff
@@ -76,14 +79,21 @@ public boolean matches(MetricID metricID, Metric metric) {
     }
 
     @Test
-    public void testMetricUpdate() {
-        // mock monitoringClient.postMetricData()
+    public void testMetricUpdate() throws InterruptedException {
+        countDownLatch1 = new CountDownLatch(1);
+        countDownLatch2 = new CountDownLatch(1);
         doAnswer(invocationOnMock -> {
+            if (countDownLatch1.getCount() != 0) {
+                // Wait for 1st signal before we can proceed
+                countDownLatch1.await(10, java.util.concurrent.TimeUnit.SECONDS);
+            } else {
+                // Give signal that 2nd call has been completed
+                countDownLatch2.countDown();
```

Comment:
Should latch 2  be counted down _after_ `testMetricUpdateCounterValue` has been updated below? 

As written, If immediately after counting down latch 2 this thread yields to the test thread, then the test code can proceed and might apply the assertion to the not-yet-updated value of  `testMetricUpdateCounterValue` leading to a false failure.

Reply:
Good point, making the change